Python/bltinmodule.c

```diff
@@ -2636,6 +2636,11 @@ zip_next(zipobject *lz)
             PyTuple_SET_ITEM(result, i, item);
             Py_DECREF(olditem);
         }
+        // The GC may have untracked this result tuple if its elements were all
+        // untracked. Since we're recycling it, make sure it's tracked again:
```

Comment:
Since this issue is quite tricky, I suggest to add a reference to `bpo-42536: `.